static/js/containers/CourseSearchPage.js

```diff
@@ -62,33 +63,28 @@ type Props = {|
 |}
 
 type State = {
-  text: string,
-  topics: Array<string>,
-  platforms: Array<string>,
-  availabilities: Array<string>,
+  text: ?string,
+  activeFacets: Map<string, Array<string>>,
   from: number,
   error: ?string
 }
 
 const shouldRunSearch = R.complement(
-  R.allPass([
-    R.eqProps("text"),
-    R.eqProps("topics"),
-    R.eqProps("platforms"),
-    R.eqProps("availabilities")
-  ])
+  R.allPass([R.eqProps("text"), R.eqProps("activeFacets")])
 )
 
 export class CourseSearchPage extends React.Component<Props, State> {
   constructor(props: Props) {
     super(props)
     this.state = {
-      text:           qs.parse(props.location.search).q,
-      topics:         qs.parse(props.location.search).t,
-      platforms:      qs.parse(props.location.search).p,
-      availabilities: qs.parse(props.location.search).a,
-      from:           0,
-      error:          null
+      text:         qs.parse(props.location.search).q,
+      activeFacets: new Map([
+        ["platform", toArray(qs.parse(props.location.search).p) || []],
+        ["topics", toArray(qs.parse(props.location.search).t) || []],
+        ["availability", toArray(qs.parse(props.location.search).a) || []]
```

Comment:
you might not want to address this now, but I was just wondering: would it make sense to store the selected options in the different facets (platform, topic, etc) as a `Set` instead of an `Array`? We can achieve the same thing roughly I suppose by using the `_.union` function and so on, but using a `Set` might make it more clear how it's supposed to be used, and make it so that we don't have to worry about de-duping and so on ourselves.

Reply:
then `activeFacets` could be a `Map<string, Set<string>>`

Reply:
Good idea, I'll do that

Reply:
Actually that seems to cause issues with using `.map(...)`, maybe best to leave as is for now.

Reply:
I just added another use of `_.union()` to ensure that there are no dupes, if you enter a URL of `../courses/search?t=Engineering&t=Engineering&type=course`, the duplicate `t=Engineering` parameter will be removed.

Reply:
yeah `Set` object do not have a `.map` method, if you want to map across them you need to first turn them into an array by doing something like `[...set]`. I don't really know why they don't have that method defined on them, seems like something you'd want to me but :woman_shrugging: